src/main/java/com/github/sylvainlaurent/maven/yamljsonvalidator/ValidateMojo.java

```diff
@@ -111,23 +114,39 @@ public void execute() throws MojoExecutionException {
 
     InputStream openJsonSchema(String jsonSchemaFile) throws MojoExecutionException {
         if (jsonSchemaFile != null && jsonSchemaFile.length() > 0) {
-            File file = new File(jsonSchemaFile);
-            if (file.isFile()) {
-                try {
-                    return new FileInputStream(file);
-                } catch (FileNotFoundException e) {
-                    throw new MojoExecutionException("Could not load schema file ["+ jsonSchemaFile + "]", e);
-                }
-            } else {
-                try {
-                    InputStream inputStream = this.getClass().getClassLoader().getResourceAsStream(jsonSchemaFile);
-                    if (inputStream != null) {
-                        return inputStream;
+            try {
```

Comment:
Could you split up these into multiple, smaller methods? It's quite hard to read it this way.